src/Model/Scope/BasicCondition.php

```diff
@@ -0,0 +1,340 @@
+    }
+
+    public function onChangeModel(): void
+    {
+        if ($model = $this->getModel()) {
+            // if we have a definitive scalar value for a field
+            // sets it as default value for field and locks it
+            // new records will automatically get this value assigned for the field
+            // @todo: consider this when condition is part of OR scope
+            if ($this->operator === '=' && !is_object($this->value) && !is_array($this->value)) {
+                // key containing '/' means chained references and it is handled in toArray method
+                if (is_string($field = $this->key) && !str_contains($field, '/')) {
+                    $field = $model->getField($field);
+                }
+
+                if ($field instanceof Field) {
+                    $field->system = true;
+                    $field->default = $this->value;
+                }
+            }
+        }
+    }
+
+    public function toArray(): array
+    {
+        // make sure clones are used to avoid changes
+        $condition = clone $this;
+
+        $field = $condition->key;
+        $operator = $condition->operator;
+        $value = $condition->value;
+
+        if ($condition->isEmpty()) {
+            return [];
+        }
+
+        if ($model = $condition->getModel()) {
+            if (is_string($field)) {
+                // shorthand for adding conditions on references
+                // use chained reference names separated by "/"
+                if (str_contains($field, '/')) {
+                    $references = explode('/', $field);
+
+                    $field = array_pop($references);
+
+                    foreach ($references as $link) {
+                        $model = $model->refLink($link);
+                    }
+
+                    // '#' -> has # referenced records
+                    // '?' -> has any referenced records
+                    // '!' -> does not have any referenced records
+                    if (in_array($field, ['#', '!', '?'], true)) {
+                        // if no operator consider '#' as 'any record exists'
+                        if ($field == '#' && !$operator) {
+                            $field = '?';
+                        }
+
+                        if (in_array($field, ['!', '?'], true)) {
+                            $operator = '=';
+                            $value = $field == '?' ? 1 : 0;
+                        }
+                    } else {
+                        // otherwise add the condition to the referenced model
+                        // and check if any records exist matching the criteria
+                        $model->addCondition($field, $operator, $value);
+                        $operator = '=';
+                        $value = 1;
+                    }
+
+                    // if not counting we check for existence only
+                    $field = $field == '#' ? $model->action('count') : $model->action('exists');
```

Comment:
`#` needs tests and optimization, this code part needs to be rechecked before merge

Reply:
Tests included